vpd-tool/src/vpd_tool_main.cpp

```diff
@@ -100,6 +123,61 @@ int main(int argc, char** argv)
         l_rc = l_vpdToolObj.readKeyword(l_vpdPath, l_recordName, l_keywordName,
                                         l_isHardwareOperation, l_filePath);
     }
+    else if (!l_writeFlag->empty())
+    {
+        std::error_code l_ec;
+
+        if (!l_hardwareFlag->empty() &&
+            !std::filesystem::exists(l_vpdPath, l_ec))
+        {
+            std::cerr << "Given EEPROM file path doesn't exist : " + l_vpdPath
+                      << std::endl;
+            return l_rc;
+        }
+        if (l_ec)
+        {
+            std::cerr << "filesystem call exists failed for file: " << l_vpdPath
+                      << ", reason: " + l_ec.message() << std::endl;
+            return l_rc;
+        }
+
+        if (!l_fileOption->empty() &&
+            !std::filesystem::exists(l_filePath, l_ec))
+        {
+            std::cerr
+                << "File doesn't exists: " << l_filePath
+                << "\nPlease provide a valid absolute file path which has keyword value.\nUse --value/--file to give "
+                   "keyword value. Refer --help."
+                << std::endl;
+            return l_rc;
+        }
+        if (l_ec)
+        {
+            std::cerr << "filesystem call exists failed for file: "
+                      << l_filePath << ", reason: " + l_ec.message()
+                      << std::endl;
+            return l_rc;
+        }
+
+        if (!l_keywordValueOption->empty() && l_keywordValue.empty())
```

Comment:
why there is no such checks for read option?
checking if given keyword is null/ give file path is null/ .. 

since these checks applies commonly for both read and write, you can make it common.

Reply:
this in write keyword condition, ready doesn't comes here.
keyword value & path checks are there already.

keeping these outside common, makes uncessary check every time for dump inv, dump object, msg clean, reset, etc.
This point is discussed already in previous PRs. "what ever the arguments required for any option will be handled inside that option handle code"